src/components/AddBook.js

```diff
@@ -1,22 +1,23 @@
 import React, { useState } from 'react';
-import { useSelector, useDispatch } from 'react-redux';
+import { useDispatch } from 'react-redux';
 import { addBook } from '../redux/books/booksSlice';
 import '../styles/AddBook.css';
 import Button from './Button';
 
 function AddBook() {
   const [title, setTitle] = useState('');
   const [author, setAuthor] = useState('');
-  const booksArr = useSelector((state) => state.book.books);
+
   const dispatch = useDispatch();
-  const clickHandler = (e) => {
-    const id = booksArr.length + 1;
+
+  const clickHandler = () => {
+    const id = Math.floor(Math.random() * 1000000);
     dispatch(addBook({
-      itemId: `item${id + 1}`,
+      item_id: `itemId${id + 1}`,
       title,
       author,
+      category: 'Action',
     }));
```

Comment:
- [ ] Great work so far. However, there's little issue with your form. You forgot to prevent form default refresh when the button type is `submit`. This has made your app to behave inconsistently. The app keep refreshing while trying to add a book due to the fact that,  default refresh is not prevented. Kindly add that to the function. It looks something similar to this. :point_down: 

```
  const clickHandler = (e) => {
      e.preventDefault();
  }
```

### App behavior
[app.webm](https://github.com/yin-ka/book-store/assets/26597432/3f36dc54-dd1e-446c-bc29-e007d0ea136b)


- [OPTIONAL] It is even better to use onSubmit prop in the form opening tag. Something like this.

```
   <form onSubmit={clickHandler}>
```

Reply:
Okay, thank you. Will work on that.